packages/ifpack2/test/unit_tests/Ifpack2_UnitTestBlockCrsUtil.hpp

```diff
@@ -300,7 +300,7 @@ struct BlockCrsMatrixMaker {
         auto* const block = &values(j*bs2);
         for (Int bi = 0; bi < bs; ++bi) {
           auto& e = block[bi*bs + bi];
-          e = 1.01*std::max(rowsum[bs*r + bi], colsum(bs*c + bi, 0))*signof(e);
+          e = Magnitude(1.01)*std::max(rowsum[bs*r + bi], colsum(bs*c + bi, 0))*signof(e);
```

Comment:
Tpetra has a function to compute one norms of rows and columns, as well as to compute diagonal entries.  You may find it useful for later revisions.  The function handles the fact that the column Map is overlapping (in reference to the question in your comment on line 281).

Reply:
This part of the code is a matrix generator for unit testing. It would be better to make unit tests as independent as possible; thus, we can isolate a problem easily.

 The reason I modified the code is that 1.01 is considered double and when complex<float> is used the right hand side result becomes complex<double> which is not possible to assign complex<float> (kokkos complex does not allow it).

Reply:
> The reason I modified the code is that 1.01 is considered double

hence these changes too: https://github.com/trilinos/Trilinos/pull/3597